script/hassfest/quality_scale_rules/runtime_data.py

```diff
@@ -0,0 +1,27 @@
+"""Validates that the configuration only uses Use ConfigEntry.runtime_data to store runtime data."""
+
+import ast
+
+from . import QualityScaleCheck
+
+
+def validate(check: QualityScaleCheck) -> None:
+    """Validate that the integration does not use hass.data."""
+
+    # Walk all files in the integration and check for hass.data
+    for file in check.integration.path.rglob("*.py"):
+        tree = ast.parse(file.read_text())
+
+        for node in ast.walk(tree):
```

Comment:
It's possible for integrations to use both hass.data (shared with all entries) AND runtime_data (entry-specific)

Reply:
Ah i didn't consider that an integration would do both, and assumed it would do one or the other and use an exemption. Thanks.

Reply:
I removed this one since it seems hard.

Reply:
Maybe enforce this for "done", and not for "exempt"

Reply:
That is already the case (we're not running any of these validators when they specify `exempt`). If we are OK using `exempt` for mixed use of `hass.data` and `runtime_data` then that works here.